hooks/open-telemetry/pkg/otel_test.go

```diff
@@ -129,20 +128,21 @@ func TestOtelHookMethods(t *testing.T) {
 		// this before hook should be blocked until the after hook for the locked resource has been run
 		go func() {
 			_, _ = otelHook.Before(openfeature.HookContext{}, openfeature.HookHints{})
-			blocked = false
+			blockedC <- struct{}{}
 		}()
-		time.Sleep(500 * time.Millisecond) // account for slow execution to ensure that the goroutine is blocked
-		if !blocked {
+		select {
+		case <-blockedC:
 			t.Fatal("duplicate keys are not being blocked")
+		default:
 		}
 
 		// unlock the resource and ensure that the previously blocked goroutine can now complete
 		otelHook.Finally(openfeature.HookContext{}, openfeature.HookHints{})
 
-		// account for slow execution time to ensure that goroutine is no longer blocked
-		// (cannot use the .Wait method in this example) as the after method has not yet been called for the blocked goroutine
-		time.Sleep(500 * time.Millisecond)
-		if blocked {
+		ticker := time.NewTicker(500 * time.Millisecond)
```

Comment:
Is this cleaner than using `time.After`?

Reply:
thanks, done